cmd/kubeadm/app/phases/controlplane/manifests_test.go

```diff
@@ -71,19 +76,30 @@ func TestGetStaticPodSpecs(t *testing.T) {
 		{
 			name:          "KubeScheduler",
 			staticPodName: kubeadmconstants.KubeScheduler,
+			env:           []v1.EnvVar{{Name: "Foo", Value: "Bar"}},
 		},
 	}
 
 	for _, tc := range tests {
 		t.Run(tc.name, func(t *testing.T) {
 			// assert the spec for the staticPodName exists
 			if spec, ok := specs[tc.staticPodName]; ok {
-
 				// Assert each specs refers to the right pod
 				if spec.Spec.Containers[0].Name != tc.staticPodName {
 					t.Errorf("getKubeConfigSpecs spec for %s contains pod %s, expects %s", tc.staticPodName, spec.Spec.Containers[0].Name, tc.staticPodName)
 				}
-
+				if tc.env != nil {
+					envs := []v1.EnvVar{}
+					// Just compare the env that testcase set as the proxy env might be set an arbitrary value on the test node.
```

Comment:
we must remove the "_proxy" env here, as the test node has the "no_proxy"env configured.

Reply:
this sounds like it needs refactor to not have tests that depend on the host env. i am talking without checking, but the host env could be passed only on actual runs / outside of tests.

Reply:
I am not sure if am following here,  correct me pls, how the host env is set seems not matter here as we just need to make sure the env set in the clusterCfg is parsed correctly.

Reply:
@neolit123 pls see the test result without the tweak here: https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/118867/pull-kubernetes-unit/1673936678231740416

Reply:
seems to me we should refactor our test here not to get:
>  [{no_proxy 127.0.0.1,localhost nil}

if i am not mistaken these come from the host?

Reply:
yes, this comes from the host. I will leave a TODO here so that we can revisit it later if we have a better way to tackle it.

Reply:
i think we can fix it here as it's relatively simple.

- make  GetStaticPodSpecs accept a []v1.EnvVar called proxyEnvs
- if the slice is nil populate it with GetProxyEnvVars() and use it in manifest construction
- if the slice is non-nil use it's value instead of GetProxyEnvVars()
- then in this unit test just pass an empty slice

does it make sense to you?

Reply:
Done, thanks for the guidance!